client/allocrunner/taskrunner/template_hook.go

```diff
@@ -116,11 +116,14 @@ func (h *templateHook) Prestart(ctx context.Context, req *interfaces.TaskPrestar
 }
 
 func (h *templateHook) Poststart(ctx context.Context, req *interfaces.TaskPoststartRequest, resp *interfaces.TaskPoststartResponse) error {
-	if req.DriverExec != nil {
+	h.managerLock.Lock()
+	defer h.managerLock.Unlock()
+
+	if req.DriverExec != nil && h.templateManager != nil {
```

Comment:
It seems like the root cause was the missing lock acquisition?

`Update` hooks [run concurrently with others](https://github.com/hashicorp/nomad/blob/v1.4.2/client/allocrunner/taskrunner/task_runner.go#L540-L543) and so if an update happens while the `Poststart` hook is running the `SetDriverHandle` call may happen between [these two lines](https://github.com/hashicorp/nomad/blob/v1.4.2/client/allocrunner/taskrunner/template_hook.go#L189-L192), causing `h.templateManager` to be `nil`.

Apart from this I'm not sure how else `templateManager` could be `nil`, should we just no-op like [in `Update`](https://github.com/hashicorp/nomad/blob/v1.4.2/client/allocrunner/taskrunner/template_hook.go#L174-L177)?

Reply:
> should we just no-op like [in Update]

I think so, yeah - as in lets check & exit on nil before entering this if/else.